### PR TITLE
NEO-946: Added new rowHeight prop for Table component

### PR DESCRIPTION
**Pull request by enrique-prado**
This prop will set the row height to either of these 3 values: compact | medium | large
large is the default value.

Also added 2 new stories and some tests
NOTE: The css classes were added to the neo-css-library repo and should now be deployed to the npm repo

image

image

### CODE REVIEW

File: src/components/Table/Table.stories.tsx [hunk 1]
```diff
@@ -342,6 +342,22 @@ export const BareBones = () => (
...
+);
+
+export const MediumRowHeight = () => (
+  <Table
+    rowHeight="medium"
+    columns={FilledFields.columns}
+    data={[...FilledFields.data]}
+  />
+);
+
```
Comment (joe-s-avaya) [inline src/components/Table/Table.stories.tsx:360]: These are unnecessary as you can just use the Templated story to show these off. We don't want to have stories for every single prop.

```suggestion
```

File: src/components/Table/Table.test.jsx [hunk 2]
```diff
@@ -32,6 +34,22 @@ describe("Table", () => {
...
+
+  it("rowHeight is set to medium ", () => {
+    const { getByRole } = render(
+      <Table rowHeight="medium" {...FilledFields} />
+    );
+
+    const tableElement = getByRole("table");
+    expect(tableElement).toHaveClass("neo-table--medium");
+  });
+
```
Comment (joe-s-avaya) [inline src/components/Table/Table.test.jsx:52]: good tests 👍

File: src/components/Table/Table.tsx [hunk 3]
```diff
@@ -181,7 +183,11 @@ export const Table = <T extends Record<string, any>>({
 
         <table
           {...getTableProps()}
-          className="neo-table"
+          className={clsx(
+            "neo-table",
+            rowHeight === "compact" && "neo-table--compact",
+            rowHeight === "medium" && "neo-table--medium"
+          )}
```
Comment (joe-s-avaya) [inline src/components/Table/Table.tsx:190]: 👍

**Review by joe-s-avaya — APPROVED** [review]
Good implementation, good defaults, good tests 👍 

I've noted that the stories are superfluous, so I suggest removing those, but otherwise this all looks really clean, nice work! 

PS, the `yarn.lock` file shouldn't have _that_ many changes to it. I'll look into why that's happening and see if I can keep that from happening again.

File: src/components/Table/Table.stories.tsx (hunk 1 above)
Comment (enrique-prado) [reply src/components/Table/Table.stories.tsx]: Good point. Removed.

**Review by mrazian85 — APPROVED** [review]
Looks good 👍

**Pull request merged**